File: DataBinding/PageAndRDLX/ODataDataSource/C#/ODataEndPoint/Controllers/MoviesController.cs
using Microsoft.Data.Sqlite;
using Microsoft.AspNetCore.OData.Routing.Controllers;
using ODataEndPoint.Models;

namespace ODataEndPoint.Controllers
{
	public class MoviesController : ODataController
	{
		private readonly IConfiguration _configuration;
		private readonly IWebHostEnvironment _env;

		public MoviesController(IConfiguration configuration, IWebHostEnvironment env)
		{
			_configuration = configuration;
			_env = env;
		}
		
		public IList<Movie> Get()
		{
			var movies = new List<Movie>();

			var connStr = _configuration.GetSection("ConnectionStrings")["Reels"]?.Replace("$appPath$", _env.ContentRootPath);
			var conn = new SqliteConnection(connStr);
			conn.Open();
			var cmd = new SqliteCommand("SELECT Movie.MovieID, Movie.Title, Movie.MPAA, Movie.YearReleased FROM Movie ORDER BY Movie.YearReleased", conn);
			var dataReader = cmd.ExecuteReader();
			while (dataReader.Read())
			{
				movies.Add(new Movie
				{
					Id = Convert.ToInt32(dataReader.GetValue(0)),
					Title = dataReader.GetValue(1).ToString()!,
					MPAA = dataReader.GetValue(2).ToString()!,
					YearReleased = Convert.ToInt32(dataReader.GetValue(3))
				});
			}
			conn.Close();
			return movies;
		}
	}
}

File: DataBinding/PageAndRDLX/ODataDataSource/VB.NET/ODataEndPoint/Controllers/MoviesController.vb
Imports Microsoft.Data.Sqlite
Imports Microsoft.AspNetCore.Hosting
Imports Microsoft.AspNetCore.OData.Routing.Controllers
Imports Microsoft.Extensions.Configuration
Imports ODataEndPoint.Models

Namespace Controllers
	Public Class MoviesController
		Inherits ODataController
		
		Dim ReadOnly _configuration As IConfiguration
		Dim ReadOnly _env As IWebHostEnvironment

		Public Sub New(configuration As IConfiguration, env As IWebHostEnvironment)
			_configuration = configuration
			_env = env
		End Sub
		
		Public Function [Get]() As IList(Of Movie)
			Dim movies = New List(Of Movie)()
			
			Dim connStr = _configuration.GetSection("ConnectionStrings")("Reels").Replace("$appPath$", _env.ContentRootPath)
			Dim conn As New SqliteConnection(connStr)
			conn.Open()
			Dim cmd As New SqliteCommand("SELECT Movie.MovieID, Movie.Title, Movie.MPAA, Movie.YearReleased FROM Movie ORDER BY Movie.YearReleased", conn)
			Dim dataReader As SqliteDatareader = cmd.ExecuteReader()
			While dataReader.Read()
				movies.Add(New Movie() With {
					          .Id = CType(dataReader.GetValue(0), Integer),
					          .Title = dataReader.GetValue(1).ToString(),
					          .MPAA = dataReader.GetValue(2).ToString(),
					          .YearReleased = CType(dataReader.GetValue(3), Integer)
					          })
			End While

			conn.Close()
			Return movies
		End Function
	End Class
End NameSpace